src/android/SimpleCameraPreview.java

```diff
@@ -173,21 +191,33 @@ private boolean disable(CallbackContext callbackContext) {
             cordova.getActivity().runOnUiThread(new Runnable() {
                 @Override
                 public void run() {
-                    webView.getView().bringToFront();
-                    webViewParent = null;
-                    FrameLayout containerView = cordova.getActivity().findViewById(containerViewId);
-                    ((ViewGroup) containerView.getParent()).removeView(containerView);
+                    try {
+                        webView.getView().bringToFront();
+                        webViewParent = null;
+                        FrameLayout containerView = cordova.getActivity().findViewById(containerViewId);
+                        ((ViewGroup) containerView.getParent()).removeView(containerView);
+                    } catch(Exception e) {
+                        e.printStackTrace();
+                        callbackContext.error(e.getMessage());
+                    }
+
                 }
             });
         }
-        fragment.disableCamera();
-        FragmentTransaction fragmentTransaction = cordova.getActivity().getFragmentManager().beginTransaction();
-        fragmentTransaction.remove(fragment);
-        fragmentTransaction.commit();
-        fragment = null;
+        try {
```

Comment:
We might have a concurency issue here.... Maybe we should put everything in the runOnUiThread or wait for the uiThread to do the rest of the process.

Reply:
@dinitri-ragoo oui, le code dans `runOnUiThread` va retirer la camera de la view et le code après le `runOnUiThread` va arrêter la caméra. Est-ce qu'on peut avoir un problème ici si le code post-`runOnUiThread` s'exécute avant que le thread puisse terminer ? Est-ce que ça fait du sens [d'attendre](https://stackoverflow.com/questions/34459392/java-wait-for-runnable-to-finish/34459442) le Runnable ?

Reply:
@ombr @azharbeebeejaun Yes, the code after the runOnUiThread is run before the one in runnable. However this does not throw any exception. I think it's because the fragment exists independently of the view. Now does it make sense to wait for the runnable ? From a theoretical view of a Fragment's LifeCycle, i think no, but if we want to be clear in our logic of the disable camera maybe we can wait for the runnable but it has no effect i think.